# Aviation Accidents Data
This is a Phase 1 Project that involves a csv file on aviation accidents between the year 1948-2022

**Author:** Emma Kahoro 
**Date:** 29/09/2025  
**Purpose:** This notebook performs a detailed analysis of aviation accident data, including accident counts, injuries and relationships with weather, months and countries.


### Data Info
The data is obtained from github as the assigned dataset [AVIATION](https://github.com/learn-co-curriculum/dsc-phase-1-project-v3/blob/master/data/Aviation_Data.csv), which contains data on aviation accidents for different aircrafts and airports.

---
The dataset contains records of aviation accidents and incidents, including the following key variables:

- **Event.Date**: Date of the accident/incident  
- **Country**: Country where the event occurred  
- **Location**: Specific location of the event  
- **Weather.Condition**: Weather at the time of the event (VMC = Visual Meteorological Conditions, IMC = Instrument Meteorological Conditions, UNK = Unknown)  
- **Investigation.Type**: Type of investigation (Accident or Incident)  
- **Total.Fatal.Injuries**: Number of fatal injuries  
- **Total.Serious.Injuries**: Number of serious injuries  
- **Total.Minor.Injuries**: Number of minor injuries  

---

## Analysis Objectives
1. Explore the distribution of accidents and incidents by **country, month and weather condition**.  
2. Analyze the **relationship between injuries and factors** such as month, weather condition and investigation type.  
3. Identify **high-risk months and countries**.  
4. Examine whether accidents are more likely to be fatal under certain conditions.  
5. Summarize findings in tables and visualizations for **insights into aviation safety**.

---



### 1.1 **Import Libraries** and **Read Dataset**
- Load Python libraries for data manipulation and visualization.
- Load the CSV file containing accident and incident records.

In [94]:
# Import Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

In [95]:
# Read the dataset
df = pd.read_csv("Aviation_Data.csv", low_memory=False)

### 1.2 Inspect the data
Taking a look at the data to get a general sense of what the DataFrame is about.
Inspect the first few rows, check column types, and understand the dataset structure. 

In [96]:
df.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude   Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN         NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN         NaN          NaN   
2    Saltville, VA  United States  36.922223  -81.878056          NaN   
3       EUREKA, CA  United States        NaN         NaN          NaN   
4       Canton, OH  United States        NaN         NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

In [97]:
df.tail()

Event.Id Investigation.Type Accident.Number  Event.Date  \
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

            Location        Country Latitude Longitude Airport.Code  \
90343  Annapolis, MD  United States      NaN       NaN          NaN   
90344    Hampton, NH  United States      NaN       NaN          NaN   
90345     Payson, AZ  United States  341525N  1112021W          PAN   
90346     Morgan, UT  United States      NaN       NaN          NaN   
90347     Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight Report.Status  \
90343             0.0               NaN                    NaN           NaN   
90344             0.0               NaN                    NaN           NaN   
90345             1.0               VMC                    NaN           NaN   
90346             0.0               NaN                    NaN           NaN   
90347             1.0               NaN                    NaN           NaN   

      Publication.Date  
90343       29-12-2022  
90344              NaN  
90345       27-12-2022  
90346              NaN  
90347       30-12-2022  

[5 rows x 31 columns]

In [98]:
print(df.shape)
print(df.columns)

(90348, 31)
Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')


In [99]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90348 entries, 0 to 90347
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      90348 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50249 non-null  object 
 9   Airport.Name            52790 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87572 non-null  object 
 14  Make                    88826 non-null

#### Summary of the Data
The general description from the df is;
- There are 90348 rows and 31 columns
- Columns have numerical and string data types
- There are missing values in most of the columns with only one column complete
- 4 columns are float datatype
- The data frame is messy and needs cleaning

### 1.3 **Data Cleaning and Transformation** 
Rectify the date datatype, handle missing values, remove duplicate values and create new columns (e.g., Year, Month).  

#### 1.3.1 Handling missing data and Duplicates

In [100]:
# See how many null values are in each column
df.isna().sum().sort_values()

# Event.Id has 1459 null values
# Drop rows where Event.Id is missing
df = df.dropna(subset=['Event.Id'])

# Check again
df.isna().sum().sort_values()


Event.Id                      0
Investigation.Type            0
Accident.Number               0
Event.Date                    0
Location                     52
Make                         63
Model                        92
Amateur.Built               102
Country                     226
Injury.Severity            1000
Registration.Number        1317
Aircraft.damage            3194
Weather.Condition          4492
Total.Uninjured            5912
Number.of.Engines          6084
Purpose.of.flight          6192
Report.Status              6381
Engine.Type                7077
Total.Fatal.Injuries      11401
Total.Minor.Injuries      11933
Total.Serious.Injuries    12510
Publication.Date          15230
Broad.phase.of.flight     27165
Airport.Name              36099
Airport.Code              38640
Latitude                  54507
Longitude                 54516
Aircraft.Category         56602
FAR.Description           56866
Air.carrier               72241
Schedule                  76307
dtype: i

In [101]:
# Check for duplicates
df.duplicated().sum()

# The DataFrame has no Duplicates

0

#### 1.3.2 Cleaning the Weather Condition Column

In [102]:
df['Weather.Condition'].value_counts()

# There is UNK and Unk both for unknown
# make the categories case sensitive
df['Weather.Condition'] = df['Weather.Condition'].str.strip().str.upper()

df['Weather.Condition'].isnull().sum()

# Replace the missing values(4492) with UNK for Unknown
df['Weather.Condition'].fillna('UNK', inplace=True)
df['Weather.Condition'].value_counts()

VMC    77303
IMC     5976
UNK     5610
Name: Weather.Condition, dtype: int64

#### 1.3.3 Changing the Event.Date column to date-time datatype

In [103]:
# Convert the columm data type do date.time 
df['Event.Date'] = pd.to_datetime(df['Event.Date'])

print(df['Event.Date'].dtype)

# Extract Year and Month(Abbreviation) and observe if they correspond correctly
df['Year'] = df['Event.Date'].dt.year
df['Month'] = df['Event.Date'].dt.strftime('%b')

df[['Year', 'Month']].head()


datetime64[ns]


Year Month
0  1948   Oct
1  1962   Jul
2  1974   Aug
3  1977   Jun
4  1979   Aug

#### 1.3.4 Clean the Location Column and the Country Column
- Drop rows where both Country and Location are missing
- Ensure the the row items are case sensitive to remove duplicates
- Replace rows with missing Location and Country to `Unknown`

In [104]:
# Count rows where both Country and Location are missing
missing_both = df[df['Country'].isnull() & df['Location'].isnull()]
print("Number of rows missing both Country and Location:", missing_both.shape[0])

# Drop rows where BOTH Country and Location are missing(1 Row)
df = df.dropna(subset=['Country', 'Location'], how='all')


Number of rows missing both Country and Location: 1


In [105]:
# Cleaning the Location Column
df['Location'].value_counts()

# Strip extra spaces and standardize to Title
df['Location'] = df['Location'].str.title().str.strip()

# Rename the missing location as `Unknown`
df['Location'].fillna('Unknown', inplace=True)

df['Location'].isna().sum() # Check for missing data

# Separate the known from the unknown locations
df_known_location = df[df["Location"] != "Unknown"]

In [106]:
# Cleaning the Country Column
df['Country'].value_counts(dropna=False).head(20)

# Strip extra spaces and standardize capitalization 
df['Country'] = df['Country'].str.strip().str.title()

# # Fill missing country values with 'Unknown'
df['Country'].fillna('Unknown', inplace=True)

df['Country'].isna().sum() # Check for missing values

# # Separate the known from the unknown Country
df_known_country = df[df["Country"] != "Unknown"]

#### 1.3.5 Data Cleaning of the Numerical Columns
Columns of interest are; `Total.Fatal.Injuries`, `Total.Serious.Injuries`, `Total.Minor.Injuries`, `Total.Uninjured`
- Drop the rows with missing values in all the columns of interest
- Ensure that all the values are numeric in nature
- Fill the missing values with 0(which is the median and the approximate mean)
- Ensure that there are no negative injuries

In [107]:
df.describe()

Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77487.000000            76378.000000   
mean            1.146585              0.647851                0.279884   
std             0.446510              5.485995                1.544093   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured          Year  
count          76955.000000     82976.000000  88888.000000  
mean               0.357066         5.325504   1999.206710  
std                2.235639        27.913797     11.888284  
min                0.000000         0.000000   1948.000000  
25%                0.000000         0.000000   1989.000000  
50%                0.000000         1.000000   1998.000000  
75%                0.000000         2.000000   2009.000000  
max              380.000000       699.000000   2022.000000

In [108]:
# Count rows where 
missing_all = df[df['Total.Fatal.Injuries'].isnull() & 
                 df['Total.Serious.Injuries'].isnull() & 
                 df['Total.Minor.Injuries'].isnull() & 
                 df['Total.Uninjured'].isnull()]

print("Number of rows missing all injuries(Fatal,Serious,Minor,Uninjured):", missing_all.shape[0])

# Drop rows where all Injuries and Uninjured are missing(223 Rows)
df = df.dropna(subset=['Total.Fatal.Injuries', 'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured'], how='all')


Number of rows missing all injuries(Fatal,Serious,Minor,Uninjured): 223


In [109]:
# # A list of the columns of interest
numeric_cols = ['Total.Fatal.Injuries', 'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured']

# #  Convert to numeric (invalid entries become NaN)
for col in numeric_cols:
    df[col] = pd.to_numeric(df[col])

# # Fill missing values with 0 (The Median and Approximate Mean)
df[numeric_cols] = df[numeric_cols].fillna(0)

# # Check for negative values in the columns
for col in numeric_cols:
    neg_count = (df[col] < 0).sum()
print(f"Number of rows with negative values: {neg_count}")

Number of rows with negative values: 0


In [110]:
# Check data type and missing data for all variables of interest
df[[
    'Country', 'Location', 'Weather.Condition', 'Event.Date',
    'Total.Fatal.Injuries', 'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured'
    ]].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 88665 entries, 0 to 90347
Data columns (total 8 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   Country                 88665 non-null  object        
 1   Location                88665 non-null  object        
 2   Weather.Condition       88665 non-null  object        
 3   Event.Date              88665 non-null  datetime64[ns]
 4   Total.Fatal.Injuries    88665 non-null  float64       
 5   Total.Serious.Injuries  88665 non-null  float64       
 6   Total.Minor.Injuries    88665 non-null  float64       
 7   Total.Uninjured         88665 non-null  float64       
dtypes: datetime64[ns](1), float64(4), object(3)
memory usage: 6.1+ MB


In [ ]:
# Save cleaned DataFrame to a new CSV
## I will put it in .gitignore
df.to_csv('Aviation_Data_Cleaned.csv', index=False)
